File: periodo-1/ip/atividades/atividade-2-beecrowd/formula-de-bhaskara.c
#include <stdio.h>
#include <math.h>

struct BhaskaraResult {
    int possible;
    double x1;
    double x2;
};

double discriminant_without_sqrt(double a, double b, double c) {
    return pow(b, 2) - (4 * a * c);
}

struct BhaskaraResult bhaskara(double a, double b, double c) {
    struct BhaskaraResult result;

    double discriminant = discriminant_without_sqrt(a, b, c);

    if (a == 0 || discriminant < 0) {
        result.possible = 0;
    } else if (discriminant == 0) {
        double square = -b / (2 * a);
        result.possible = 1;
        result.x1 = square;
        result.x2 = square;
    } else {
        result.possible = 1;
        result.x1 = (-b + sqrt(discriminant)) / (2 * a);
        result.x2 = (-b - sqrt(discriminant)) / (2 * a);
    }

    return result;
}

int main() {
    double a, b, c, x1, x2;

    scanf("%lf %lf %lf", &a, &b, &c);

    struct BhaskaraResult result = bhaskara(a, b, c);

    if (result.possible == 1) {
        printf("R1 = %.5lf\n", result.x1);
        printf("R2 = %.5lf\n", result.x2);
    } else {
        printf("Impossivel calcular\n");
    }

    return 0;
}

File: periodo-1/ip/atividades/atividade-4-beecrowd/formula_de_bhaskara.py
from math import sqrt

def main():
    a,b,c = [float(x) for x in input().split(" ")]

    if (a == 0):
        print("Impossivel calcular")
        return

    def calcular_delta(a, b, c) -> float:
        return (b ** 2) - 4 * a * c

    def calcular_bhaskara_duas_raizes(a, b, delta) -> dict:
        return {
            "x1": (-b + sqrt(delta)) / (2 * a),
            "x2": (-b - sqrt(delta)) / (2 * a)
        }

    delta = calcular_delta(a, b, c)

    if (delta < 0):
        print("Impossivel calcular")
        return
    
    raizes = calcular_bhaskara_duas_raizes(a, b, delta)

    print(f"R1 = {raizes.get('x1'):.5f}")
    print(f"R2 = {raizes.get('x2'):.5f}")

main()
